### PR TITLE
Comply to YAML spec where possible

### PR DESCRIPTION
**Pull request by mike-nguyen**
This commit fixes minor issues in the atomic host tests to comply
to the YAML spec.  The issues fixed were found using yamllint.

There were some instances of line length that could not be fixed
(i.e. Ansible when statements like  "'foo' in bar"). Using > or |
would break the logic so the line length had to be preserved.

### CODE REVIEW

File: tests/admin-unlock/main.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@
 
 - name: Ostree Admin Unlock - Setup
   hosts: all
-  become: yes
+  become: "yes"
```
Comment (cgwalters) [inline tests/admin-unlock/main.yml]: Weird, why?

Comment (mike-nguyen) [reply tests/admin-unlock/main.yml]: yamllint complains -- warning  truthy value is not quoted  (truthy)

This is the rule: http://yamllint.readthedocs.io/en/latest/rules.html#module-yamllint.rules.truthy

Comment (cgwalters) [reply tests/admin-unlock/main.yml]: Ah and I see: http://yaml.org/type/bool.html

But something I don't quite understand here; the Ansible docs say the `become` flag is a boolean, so we shouldn't be converting it to a string AIUI.  

Or oh ugh, do we really need `!!bool true`?

Comment (miabbott) [reply tests/admin-unlock/main.yml]: I think we can avoid the quoting by just using `become: true`

Comment (miabbott) [reply tests/admin-unlock/main.yml]:
```
$ cat test.yml 
---
become: true
$ yamllint test.yml && echo $?
0
```

Comment (miabbott) [reply tests/admin-unlock/main.yml]: This is in the same area as #299

Comment (mike-nguyen) [reply tests/admin-unlock/main.yml]: I'll move it over to true/false

Comment (cgwalters) [reply tests/admin-unlock/main.yml]: Ah true, looks like the same issue.

File: tests/pkg-layering/main.yml [hunk 2]
```diff
@@ -235,21 +244,25 @@
...
+        pending_json['deployments'][0] is defined
+        and not pending_json['deployments'][0]['booted']
 
     - name: Set packages when deployment 1 is not booted
       set_fact:
         pending_pkgs: "{{ pending_json['deployments'][1]['packages'] }}"
-      when: pending_json['deployments'][1] is defined and not pending_json['deployments'][1]['booted']
+      when: >
+        pending_json['deployments'][1] is defined
+        and not pending_json['deployments'][1]['booted']
```
Comment (miabbott) [inline tests/pkg-layering/main.yml]: My preference is to use a list for these kind of logical ANDs, but this is A-OK

```
when:
  - pending_json['deployments'][1] is defined
  - not pending_json['deployments'][1]['booted']
```

File: tests/pkg-layering/main.yml [hunk 3]
```diff
@@ -235,21 +244,25 @@
...
 
-    - name: Fail if {{ g_pkg2 }} is in rpm-ostree status output for pending packages
+    - name: Fail if {{ g_pkg2 }} is in  pending packages
```
Comment (miabbott) [inline tests/pkg-layering/main.yml]: nit: extra space between `in` and `pending`

**Comment by miabbott** [comment]
A couple of minor nits, but this is overall OK.  I'd like to understand why the CI checks are failing because the failures don't look related to these changes.

**Comment by mike-nguyen** [comment]
bot, retest this please

**Comment by miabbott** [comment]
@mike-nguyen have you had a chance to look at these failures?

**Comment by miabbott** [comment]
CentOS failures look like #316 

Fedora 26 failure due to `docker-latest` not being installed

Fedora 27 failure is infra flake


I don't see any huge functional changes here, so I'm inclined to merge this rather than fighting unrelated errors.

**Comment by miabbott** [comment]
@mike-nguyen Feel free to merge based on comment ⬆️

**Comment by mike-nguyen** [comment]
@miabbott I have all the tests running on RHELAH right now.  Once that passes I will merge.

**Comment by mike-nguyen** [comment]
Merging this

**Pull request merged**